src/Input/AbstractInputParam.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Input;
+
+use InvalidArgumentException;
+use Symfony\Component\Console\Input\InputOption;
+
+use function array_walk;
+use function in_array;
+use function is_array;
+use function is_string;
+use function sprintf;
+
+/**
+ * Provide the majority of methods needed to implement InputParamInterface.
+ *
+ * This trait provides definitions for all but the following methods of the
+ * InputParamInterface:
+ *
+ * - getOptionMode()
+ * - getQuestion()
+ *
+ * Additionally, it defines the `$name` property, allowing implementations to
+ * set it in their constructors without needing to define the property
+ * themselves.
+ */
+abstract class AbstractInputParam implements InputParamInterface
+{
+    /** @var int[] */
+    private $allowedModes = [
+        InputOption::VALUE_NONE,
+        InputOption::VALUE_REQUIRED,
+        InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+    ];
+
+    /** @var mixed */
+    private $default;
+
+    /** @var string */
+    private $description = '';
+
+    /**
+     * Parameter name; must be set by class composing trait!
+     *
+     * @var string
+     */
+    private $name;
+
+    /**
+     * InputOption mode to use with this parameter.
+     *
+     * @var int
+     */
+    private $optionMode = InputOption::VALUE_REQUIRED;
+
+    /** @var bool */
+    private $required = false;
+
+    /** @var null|string|array */
+    private $shortcut;
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * Default value to use if none provided.
+     *
+     * @return mixed
+     */
+    public function getDefault()
+    {
+        return $this->default;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getOptionMode(): int
+    {
+        return $this->optionMode;
+    }
+
+    // phpcs:disable WebimpressCodingStandard.Functions.ReturnType.ReturnValue
```

Comment:
It inherits that from the interface; I've had to provide rules like this elsewhere for the same reason. I'm changing it to `phpcs:ignore` instead of `phpcs:disable`, as that way it will continue to check throughout the file, and only ignore the one line.

However, if we add a docblock, we risk it going out-of-date with the interface, and also create potential conflicts/problems for phpstan/psalm if/when we add those tools.

Reply:
Odd - I ended up having to add it after all, because phpcs didn't like `phpcs:ignore`. :shrug: